### PR TITLE
VPN-4690: Add excluded apps help sheet

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## Description

**NOTE: Depends on [#8864: VPN-4687: Add DNS settings help sheet](https://github.com/mozilla-mobile/mozilla-vpn-client/pull/8864)**

- Implement help sheet for excluded apps

https://github.com/mozilla-mobile/mozilla-vpn-client/assets/15353801/bfde51ba-a118-4824-ad77-edccd191f8da

## Reference

[VPN-4690: Implement the 'App Exclusions' in-product help sheet](https://mozilla-hub.atlassian.net/browse/VPN-4690)

### CODE REVIEW

File: src/translations/strings.yaml [hunk 1]
```diff
@@ -928,6 +928,25 @@ devices:
...
     comment: Title for the menu bar when viewing the devices or device limit pages
 
+  excludedAppsTitle:
+    value: Excluded apps
+    comment: Title label for the excluded apps help sheet
+  excludedAppsHeader:
+    value: What are excluded apps? 
+    comment: Header label for the excluded apps help sheet
+  excludedAppsBody1:
+    value: Excluded apps let you turn off VPN protection for specific apps, without turning off your device’s VPN protection.
```
Comment (flodolo) [inline src/translations/strings.yaml:938]: Is it just me, or this can be confusing? It makes it sound like the device has a VPN protection that is unrelated to Mozilla VPN, unlike something like this:

> … without turning off VPN protection for your entire device.

File: src/translations/strings.yaml [hunk 2]
```diff
@@ -928,6 +928,25 @@ devices:
...
+    comment: Body label for the excluded apps help sheet
+  excludedAppsBody2:
+    value: When you exclude an app, your IP address and approximate location will be exposed to it.
```
Comment (flodolo) [inline src/translations/strings.yaml:941]:
```suggestion
    value: When you exclude an app, it will have access to your IP address and approximate location.
```

In case content needs to check these strings, this feels a bit easier to read.

**Review by flodolo — APPROVED** [review]
A couple of questions for English content, only the first really affecting localization.

File: src/translations/strings.yaml (hunk 1 above)
Comment (MattLichtenstein) [reply src/translations/strings.yaml:938]: @traceyvantyghem

File: src/translations/strings.yaml (hunk 2 above)
Comment (MattLichtenstein) [reply src/translations/strings.yaml:941]: @traceyvantyghem

File: src/translations/strings.yaml (hunk 1 above)
Comment (traceyvantyghem) [reply src/translations/strings.yaml:938]: Sure, we can make this change. Let's update to Flod's suggestion:

Excluded apps let you turn off VPN protection for specific apps, without turning off VPN protection for your entire device.

File: src/translations/strings.yaml (hunk 2 above)
Comment (traceyvantyghem) [reply src/translations/strings.yaml:941]: Sure, I also like this suggestion. 

Thanks Flod.

File: src/mozillavpn.cpp [hunk 3]
```diff
@@ -1381,6 +1381,11 @@ void MozillaVPN::registerUrlOpenerLabels() {
   uo->registerUrlLabel("upgradeToAnnualUrl", []() -> QString {
     return Constants::upgradeToAnnualUrl();
   });
+
+  uo->registerUrlLabel("sumoExcludedApps", []() -> QString {
+    return "https://support.mozilla.org/en-US/kb/"
```
Comment (lesleyjanenorton) [inline src/mozillavpn.cpp:1386]: let's drop the "en-US" here because it stipulates that the page is shown in English. Without it, moz.org will route the page based on the person's browser language prefs.

File: src/ui/screens/settings/appPermissions/ViewAppPermissions.qml [hunk 4]
```diff
@@ -64,6 +83,32 @@ MZViewBase {
...
+
+        active: false
+
+        onActiveChanged: if (active) item.open()
+
+        sourceComponent: MZHelpSheet {
+            title: MZI18n.HelpSheetsExcludedAppsTitle
+
+            model: [
+                {type: MZHelpSheet.BlockType.Title, text: MZI18n.HelpSheetsExcludedAppsHeader},
```
Comment (lesleyjanenorton) [inline src/ui/screens/settings/appPermissions/ViewAppPermissions.qml:97]: nit: magic numbers

File: src/ui/screens/settings/appPermissions/ViewAppPermissions.qml [hunk 5]
```diff
@@ -64,6 +83,32 @@ MZViewBase {
...
+                {type: MZHelpSheet.BlockType.Text, text: MZI18n.HelpSheetsExcludedAppsBody1, margin: 8},
+                {type: MZHelpSheet.BlockType.Text, text:MZI18n.HelpSheetsExcludedAppsBody2, margin: 16},
```
Comment (lesleyjanenorton) [inline src/ui/screens/settings/appPermissions/ViewAppPermissions.qml:99]: > text:MZI18n.HelpSheetsExcludedAppsBody2

nit: space between 'text:' and 'MZ...'

**Review by lesleyjanenorton — CHANGES_REQUESTED** [review]
This looks great! 🤩 One thing to change in the URL but other comments are just nits.

File: src/ui/screens/settings/appPermissions/ViewAppPermissions.qml (hunk 4 above)
Comment (MattLichtenstein) [reply src/ui/screens/settings/appPermissions/ViewAppPermissions.qml:97]: will fix for all help sheets in a fast follow

**Review by lesleyjanenorton — APPROVED** [review]

**Review by vinocher — APPROVED** [review]
